ggml/src/ggml-aarch64.c

```diff
@@ -351,6 +354,28 @@ size_t quantize_q4_0_8x8(const float * restrict src, void * restrict dst, int64_
     return quantize_q4_0_nr_bl(src, dst, nrow, n_per_row, 8, 8);
 }
 
+// Print a given message only once
+static inline void print_message_once(const char* message) {
+    static bool printed = false;
+    if (!printed) {
+        fprintf(stderr, "%s\n", message);
+        printed = true;
+    }
+}
+
+// Return The number of 32-bit lanes in the SVE vector if SVE is supported; otherwise, returns 0 if SVE is not supported.
+static int sve_lane_count(void) {
+    static int lane_count = -1;
+    if (lane_count == -1) {
+#if defined(__ARM_FEATURE_SVE)
+        lane_count = ggml_sve_cnt_b;
+#else
+        lane_count = 0;
+#endif
+    }
+    return lane_count;
+}
+
```

Comment:
I think these will generate data race warnings when the thread sanitizer is enabled.

Reply:
@ggerganov Thanks for reviewing the code and I'll update to make the function thread safe.

Reply:
@ggerganov I have pushed the commit that addresses your concern about the data race condition.